code/modules/sprite_accessories/_accessory.dm

```diff
@@ -21,8 +21,27 @@
 	var/icon                                       // the icon file the accessory is located in
 	var/icon_state                                 // the icon_state of the accessory
 	var/gender = null                              // Restricted to specific genders. null matches any
-	var/list/species_allowed = list(SPECIES_HUMAN) // Restrict some styles to specific bodytypes
-	var/list/subspecies_allowed                    // Restrict some styles to specific species names
+	var/list/species_allowed = list(SPECIES_HUMAN) // Restrict some styles to specific root species names
+	var/list/subspecies_allowed                    // Restrict some styles to specific species names, irrespective of root species name
+	var/bodytypes_allowed = null                   // Restrict some styles to specific bodytypes
+	var/bodytypes_denied =  null                   // Restrict some styles to specific bodytypes
 	var/do_colouration = 1                         // Whether or not the accessory can be affected by colouration
 	var/blend = ICON_ADD
-	var/flags = 0
\ No newline at end of file
+	var/flags = 0
+
+/decl/sprite_accessory/proc/accessory_is_available(var/mob/owner, var/decl/species/species, var/bodytype_flags, var/check_gender)
+	if(gender && check_gender != gender)
```

Comment:
Not sure this is actually the logic you intended: you might also want it to succeed if `check_gender` is null (or false?).